config/initializers/event_job_initializer.rb

```diff
@@ -22,14 +22,28 @@ def collection_event(col)
     end
 
     def log_to_all_followers(col, obj, object_removal=false)
+      if !col.collections.empty?
+        parents = col.collections
+      end
+      
       col.followers.each do |user|
         if !object_removal
           next unless user.can?(:read, obj)
         end
+        
         next unless user.can?(:read, col)
-        user.log_event(collection_event(col))
+
+        if parents.nil?
+          user.log_event(collection_event(col))
+        else 
+          parents.each do |parent|
+            if !parent.followers.include?(user)
```

Comment:
This has a potential to spam. What if there were 5 parents? Would it log the event for `user` 5 times? I think to do what you're looking for, you need to get followers of all parent collections at once. Something like this would be enough to replace all the code you've added to this method and fix the problem:
```ruby
user.log_event(collection_event(col)) if !col.collections.map {|o| o.followers }.flatten.include?(user)
```

Reply:
i did not account for that. forgot that collections and generic files can belong to more than one collections. will update this. thanks for the code.